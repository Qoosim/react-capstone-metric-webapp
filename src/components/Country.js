import React from 'react';
import PropTypes from 'prop-types';

const Country = ({ name, info }) => {
  const {
    today_confirmed: confirmedTotal,
    today_deaths: deathsTotal,
    today_recovered: recoveredTotal,
  } = info;

  return (
    <>
      <div>
        <h2>{name}</h2>
        <p>
          Confirmed:
          {' '}
          {confirmedTotal}
        </p>
        <p>
          Deaths:
          {' '}
          {deathsTotal}
        </p>
        <p>
          Recovered:
          {' '}
          {recoveredTotal}
        </p>
      </div>
    </>
  );
};

Country.propTypes = {
  name: PropTypes.string.isRequired,
  info: PropTypes.shape({
    id: PropTypes.string,
    date: PropTypes.string,
    today_confirmed: PropTypes.string,
    today_deaths: PropTypes.string,
    today_recovered: PropTypes.string,
    new_confirmed: PropTypes.string,
    new_deaths: PropTypes.string,
    today_new_cases: PropTypes.string,
    today_new_recovered: PropTypes.string,
    today_open_cases: PropTypes.string,
    yesterday_confirmed: PropTypes.string,
    yesterday_deaths: PropTypes.string,
    yesterday_open_cases: PropTypes.string,
    yesterday_recovered: PropTypes.string,
  }).isRequired,
};

export default Country;
