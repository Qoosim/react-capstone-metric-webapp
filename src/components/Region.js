import PropTypes from 'prop-types';
import { Card, Col, Row } from 'react-bootstrap';

const Region = ({ name, info }) => {
  const {
    today_confirmed: confirmedTotal,
    today_deaths: deathsTotal,
    today_recovered: recoveredTotal,
  } = info;

  return (
    <Card className="rounded-0 h-100 bg-region" data-testid="region">
      <Card.Body className="d-flex flex-column justify-content-center">
        <Row xs={2}>
          <Card.Title className="mb-0 d-flex flex-column justify-content-center">{name}</Card.Title>
          <Col>
            <Row xs={1}>
              <Col>
                Confirmed:
                {' '}
                {confirmedTotal}
              </Col>
              <Col>
                Deaths:
                {' '}
                {deathsTotal}
              </Col>
              <Col>
                Recovered:
                {' '}
                {recoveredTotal}
              </Col>
            </Row>
          </Col>
        </Row>
      </Card.Body>
    </Card>
  );
};

Region.propTypes = {
  name: PropTypes.string.isRequired,
  info: PropTypes.shape({
    today_confirmed: PropTypes.number,
    today_deaths: PropTypes.number,
    today_recovered: PropTypes.number,
  }).isRequired,
};

export default Region;
