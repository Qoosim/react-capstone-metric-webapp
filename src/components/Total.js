import PropTypes from 'prop-types';
import { Col, Row } from 'react-bootstrap';

const Total = ({
  confirmedTotal, deathsTotal,
  recoveredTotal, confirmedToday,
  deathsToday, recoveredToday,
}) => (
  <Row xs={1} className="mx-0 gy-1 mb-2">
    <Col>
      Total confirmed:
      {' '}
      {confirmedTotal}
    </Col>
    <Col className="bg-custom3">
      Total deaths:
      {' '}
      {deathsTotal}
    </Col>
    <Col>
      Total recovered:
      {' '}
      {recoveredTotal}
    </Col>
    <Col className="bg-custom3">
      Today&apos;s new confirmed:
      {' '}
      {confirmedToday}
    </Col>
    <Col>
      Today&apos;s new deaths:
      {' '}
      {deathsToday}
    </Col>
    <Col className="bg-custom3">
      Today&apos;s new recovered:
      {' '}
      {recoveredToday}
    </Col>
  </Row>
);

Total.propTypes = {
  confirmedTotal: PropTypes.number.isRequired,
  deathsTotal: PropTypes.number.isRequired,
  recoveredTotal: PropTypes.number.isRequired,
  confirmedToday: PropTypes.number.isRequired,
  deathsToday: PropTypes.number.isRequired,
  recoveredToday: PropTypes.number.isRequired,
};

export default Total;
