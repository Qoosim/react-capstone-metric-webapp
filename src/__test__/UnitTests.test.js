import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import Country from '../components/Country';
import Region from '../components/Region';
import Search from '../components/Search';
import Total from '../components/Total';

const [countryName, countryRegion,
  confirmedTotal, deathsTotal,
  recoveredTotal, index,
] = [
  'Argentina',
  'Mendoza',
  200,
  10,
  112,
  0,
];

const covidDataTotalMock = {
  today_confirmed: 210,
  today_deaths: 10,
  today_recovered: 112,
  today_new_confirmed: 2,
  today_new_deaths: 0,
  today_new_recovered: 10,
};

describe('testing snapshots for the child components', () => {
  test('testing snapshot for Country component', () => {
    const tree = render(
      <Router>
        <Country
          index={index}
          name={countryName}
          confirmedTotal={confirmedTotal}
          deathsTotal={deathsTotal}
          recoveredTotal={recoveredTotal}
        />
      </Router>,
    );
    expect(tree).toMatchSnapshot();
  });

  test('testing snapshot for Region component', () => {
    const tree = render(
      <Router>
        <Region
          index={index}
          name={countryRegion}
          confirmedTotal={confirmedTotal}
          deathsTotal={deathsTotal}
          recoveredTotal={recoveredTotal}
        />
      </Router>,
    );
    expect(tree).toMatchSnapshot();
  });

  test('testing snapshot for Total component', () => {
    const tree = render(
      <Total
        covidDataTotal={covidDataTotalMock}
      />,
    );
    expect(tree).toMatchSnapshot();
  });

  test('testing snapshot for Search component', () => {
    const tree = render(
      <Search
        value=""
        onChange={() => {}}
        clear={() => {}}
      />,
    );
    expect(tree).toMatchSnapshot();
  });
});
