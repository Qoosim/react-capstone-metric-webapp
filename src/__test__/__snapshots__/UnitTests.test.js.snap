// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing snapshots for the child components testing snapshot for Country component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="rounded-0 h-100 cursor-pointer bg-country card"
        data-testid="country"
      >
        <div
          class="d-flex flex-column justify-content-center card-body"
        >
          <div
            class="card-title h5"
          >
            Argentina
          </div>
          <div
            class="row row-cols-1"
          >
            <div
              class="col"
            >
              Confirmed:
               
              200
            </div>
            <div
              class="col"
            >
              Deaths:
               
              10
            </div>
            <div
              class="col"
            >
              Recovered:
               
              112
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="rounded-0 h-100 cursor-pointer bg-country card"
      data-testid="country"
    >
      <div
        class="d-flex flex-column justify-content-center card-body"
      >
        <div
          class="card-title h5"
        >
          Argentina
        </div>
        <div
          class="row row-cols-1"
        >
          <div
            class="col"
          >
            Confirmed:
             
            200
          </div>
          <div
            class="col"
          >
            Deaths:
             
            10
          </div>
          <div
            class="col"
          >
            Recovered:
             
            112
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`testing snapshots for the child components testing snapshot for Region component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="rounded-0 h-100 bg-region card"
        data-testid="region"
      >
        <div
          class="d-flex flex-column justify-content-center card-body"
        >
          <div
            class="row row-cols-2"
          >
            <div
              class="mb-0 d-flex flex-column justify-content-center card-title h5"
            >
              Mendoza
            </div>
            <div
              class="col"
            >
              <div
                class="row row-cols-1"
              >
                <div
                  class="col"
                >
                  Confirmed:
                   
                  200
                </div>
                <div
                  class="col"
                >
                  Deaths:
                   
                  10
                </div>
                <div
                  class="col"
                >
                  Recovered:
                   
                  112
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="rounded-0 h-100 bg-region card"
      data-testid="region"
    >
      <div
        class="d-flex flex-column justify-content-center card-body"
      >
        <div
          class="row row-cols-2"
        >
          <div
            class="mb-0 d-flex flex-column justify-content-center card-title h5"
          >
            Mendoza
          </div>
          <div
            class="col"
          >
            <div
              class="row row-cols-1"
            >
              <div
                class="col"
              >
                Confirmed:
                 
                200
              </div>
              <div
                class="col"
              >
                Deaths:
                 
                10
              </div>
              <div
                class="col"
              >
                Recovered:
                 
                112
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`testing snapshots for the child components testing snapshot for Search component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mx-0 gx-0 px-2 row"
      >
        <div
          class="col"
        >
          <input
            class="form-control"
            placeholder="Search..."
            value=""
          />
        </div>
        <div
          class="col-auto"
        >
          <button
            class="btn btn-danger"
            type="button"
          >
            Clear
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mx-0 gx-0 px-2 row"
    >
      <div
        class="col"
      >
        <input
          class="form-control"
          placeholder="Search..."
          value=""
        />
      </div>
      <div
        class="col-auto"
      >
        <button
          class="btn btn-danger"
          type="button"
        >
          Clear
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`testing snapshots for the child components testing snapshot for Total component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mx-0 gy-1 mb-2 row row-cols-1"
      >
        <div
          class="col"
        >
          Total confirmed:
           
          210
        </div>
        <div
          class="bg-custom3 col"
        >
          Total deaths:
           
          10
        </div>
        <div
          class="col"
        >
          Total recovered:
           
          112
        </div>
        <div
          class="bg-custom3 col"
        >
          Today's new confirmed:
           
          2
        </div>
        <div
          class="col"
        >
          Today's new deaths:
           
          0
        </div>
        <div
          class="bg-custom3 col"
        >
          Today's new recovered:
           
          10
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mx-0 gy-1 mb-2 row row-cols-1"
    >
      <div
        class="col"
      >
        Total confirmed:
         
        210
      </div>
      <div
        class="bg-custom3 col"
      >
        Total deaths:
         
        10
      </div>
      <div
        class="col"
      >
        Total recovered:
         
        112
      </div>
      <div
        class="bg-custom3 col"
      >
        Today's new confirmed:
         
        2
      </div>
      <div
        class="col"
      >
        Today's new deaths:
         
        0
      </div>
      <div
        class="bg-custom3 col"
      >
        Today's new recovered:
         
        10
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
